### PR TITLE
[v2.6.3 patch1] Include changes of webhook certificate auto-renewal

### PR DESCRIPTION
**Pull request by maxsokolovsky**
Related to https://github.com/rancher/rancher/issues/36177.

Cherry-pick commits from tag `v0.2.3` until and including the dynamic listener update commit – to enable automatic cert renewal.
Once this PR is merged, I will tag the resulting latest commit as `v0.2.4` (to be used only in the `v2.6.3-patch1` release).

### CODE REVIEW

**Review by cmurphy — COMMENTED** [review]
Are you sure the CAPI changes in b912e8c6034529c12716e63e4edc2d692cbc6ef2 are meant to be included in 2.6.3-patch1? This would have an impact on QA.

**Comment by maxsokolovsky** [comment]
> Are you sure the CAPI changes in [b912e8c](https://github.com/rancher/webhook/commit/b912e8c6034529c12716e63e4edc2d692cbc6ef2) are meant to be included in 2.6.3-patch1? This would have an impact on QA.

@cmurphy, good point. I included it because the latest relevant commit was created after that CAPI bump one, so I thought, it may be needed for my latest commit. My plan was to include everything from the tag `v0.2.3` until my change, inclusively.

I am not 100% certain that my commit would have completely functional code without that earlier CAPI bump. By looking at it, I don't think that bump affects it. But may have to test first locally. Should we remove that commit?

**Comment by thedadams** [comment]
@maxsokolovsky The CAPI stuff should not be in 2.6.3-patch1. Your dependency bump shouldn't require those changes.

**Comment by maxsokolovsky** [comment]
Thanks, updated the PR, ready for review.

**Review by cmurphy — APPROVED** [review]
Thanks!

File: go.mod
Comment (cbron) [inline go.mod:33]: > Once this PR is merged, I will tag the resulting latest commit as v0.2.4 (to be used only in the v2.6.3-patch1 release).

Are we going to release this tagged as 0.2.4, or as 2.6.3-patch1

**Review by cbron — COMMENTED** [review]
Looks ok to me. Hard to tell with wrangler but both of those 0.8.8 and 0.8.9 releases look fairly unrelated. One question on tagging.

File: go.mod
Comment (maxsokolovsky) [reply go.mod:33]: I had checked before opening the PR with @aiyengar2. We'd prefer not to disturb the `v0.2.4` series of chart RCs if possible.
Assuming we can have a chart with `2.6.3-patch1` as the version, we'd choose to go that route?

Comment (cbron) [reply go.mod:33]: I think that makes sense

**Review by cbron — APPROVED** [review]
lgtm, if @thedadams is around maybe he can also approve the patch tag approach (see convo above).

File: go.mod
Comment (thedadams) [reply go.mod:33]: @maxsokolovsky @cbron We should not tag this release of rancher-webhook as `2.6.3-patch1`:

Anyone that updates to Rancher v2.6.3-patch1 would then get rancher-webhook `v2.6.3-patch1`. If they then tried to update to Rancher v2.6.4 when it is released, then the logic in Rancher would keep rancher-webhook version `v2.6.3-patch1` because this version is greater than the `MIN_WEBHOOK_VERSION` in Rancher v2.6.4 (which will be `v0.2.4`).

The problem with this is that rancher-webhook `v0.2.4` will have important updates in it related to CAPI that `v2.6.3-patch1` will not have. This is likely to break Rancher.

**Review by thedadams — APPROVED** [review]
As long as we don't tag this version of rancher-webhook as `v2.6.3-patch1`, then I am good with these changes.

**Comment by cbron** [comment]
Thats exactly what we were going to do 😄

File: go.mod
Comment (cbron) [reply go.mod:33]: Sorry we got mixed up on versions, we are going to tag this as 0.2.3-patch1. So you have: 
2.6.3: webhook 0.2.3
2.6.3-patch1: webhook 0.2.3-patch1
2.6.4: webhook 0.2.4
Going that route we should be fine, as 0.2.4 is greater than 0.2.3-patch1.

**Pull request merged**
